src/diff/index.js

```diff
@@ -438,7 +438,11 @@ function diffElementNodes(
 			if (
 				'value' in newProps &&
 				(i = newProps.value) !== undefined &&
-				i !== dom.value
+				// #2756 For the <progress>-element the initial value is 0,
+				// despite the attribute not being present. When the attribute
+				// is missing the progress bar is treated as indeterminate.
+				// To fix that we'll always update it when it is 0 for progress elements
+				(i !== dom.value || (newVNode.type === 'progress' && !i))
```

Comment:
Turns out my previous fix of doing `!dom.hasAttribute('value')` is not a good one as it leads to all input values always being updated. The new fix only updates it, despite `dom.value === props.value` when the value is `0` for progress elements.